### PR TITLE
Serialize container image builds based on dependencies.

### PR DESCRIPTION
**Pull request by rtnpro**
Run dependent container builds in sequence.

### CODE REVIEW

File: beanstalk_worker/Dockerfile.build_worker [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN yum update -y && \
     yum clean all
 
 RUN mkdir -p /run_worker
-ADD beanstalkc.py ca.crt node.kubeconfig worker_start_build.py config.py /run_worker/
+ADD beanstalkc.py ca.crt node.kubeconfig worker_start_build.py config.py lib.py /run_worker/
```
Comment (mohammedzee1000) [inline beanstalk_worker/Dockerfile.build_worker]: I am assuming the lib.py is a mock of some sort as its not in repo right now, nor is it in this PR.

**Comment by mohammedzee1000** [comment]
@rtnpro can you rebase please. It has conflicts

File: beanstalk_worker/worker_start_build.py [hunk 2]
```diff
@@ -77,7 +78,7 @@ def start_build(job_details):
         appid = job_details['appid']
         jobid = job_details['jobid']
         desired_tag = job_details['desired_tag']
-        namespace = str(appid) + "-" + str(jobid) + "-" + str(desired_tag)
+        namespace = get_job_name(job_details)
```
Comment (bamachrn) [inline beanstalk_worker/worker_start_build.py]: 👍

File: beanstalk_worker/worker_start_build.py [hunk 3]
```diff
@@ -165,17 +170,34 @@ def start_build(job_details):
...
+
+                if parent_build_running:
+                    logger.debug('Parents in build: %s, pushing job: %s back '
+                                 'to queue' % (parents_in_build, job_details))
+                    bs.use('master_tube')
+                    bs.put(json.dumps(job_details))
+                    bs.use('failed_build')
+                else:
+                    logger.debug(str(job_details))
+                    result = start_build(job_details)
```
Comment (bamachrn) [inline beanstalk_worker/worker_start_build.py]: we are not touching some file to mark build started? may be use Build.start()

**Comment by bamachrn** [comment]
@rtnpro you want to rebase this?

**Pull request merged**